arch.mk

```diff
@@ -233,15 +270,21 @@ endif
 
 CFLAGS+=-DARCH_FLASH_OFFSET=$(ARCH_FLASH_OFFSET)
 
-## Toolchain setup
-CC=$(CROSS_COMPILE)gcc
-LD=$(CROSS_COMPILE)gcc
-AS=$(CROSS_COMPILE)gcc
-OBJCOPY:=$(CROSS_COMPILE)objcopy
-SIZE:=$(CROSS_COMPILE)size
-BOOT_IMG?=test-app/image.bin
+# Setup default optimizations (for GCC)
```

Comment:
without defaulting on `$(CROSS_COMPILE)gcc`, I get `CC=cc` as default on my system, so it won't invoke the cross compiler properly.

```
make V=1
	[CC-ARM] hal/stm32f4.o
cc -D"__WOLFBOOT" -D"WOLFBOOT_VERSION=0UL" [...]
```

Please restore $(CROSS_COMPILE)gcc as default choice, or the CC variable will have "cc" in it